src/python/zquantum/qaoa/estimators/_cvar_estimator.py

```diff
@@ -175,11 +175,24 @@ def _sum_expectation_values(
     return final_value
 
 
-def _calculate_expectation_value_of_bitstring(
-    bitstring: str, operator: IsingOperator
-) -> float:
-    """Calculate expectation value for a bitstring based on an operator."""
-    expected_value = Measurements([bitstring]).get_expectation_values(
-        operator, use_bessel_correction=False
-    )
-    return np.sum(expected_value.values)
+def _calculate_expectation_values(
+    bitstrings: np.ndarray, operator: IsingOperator
+) -> np.ndarray:
+    """Calculates expectation values for each bitstring in the given array"""
+
+    if not isinstance(operator, IsingOperator):
+        raise TypeError("Input operator not openfermion.IsingOperator")
+
+    expectation_values_list = [
+        coefficient
+        * (
+            check_parity_of_vector(
+                bitstrings,
+                [cast(int, op[0]) for op in term],
```

Comment:
[question] I don't get this part – why are we casting to int here and what's exactly under `op[0]`?

Reply:
Actually I don't think we need to cast to int, good observation. `op[0]` is the qubit of the hamiltonian term, because openfermion represents operators like `(0, 'Z')`.

Reply:
Fixed in [38e4540](https://github.com/zapatacomputing/z-quantum-qaoa/pull/88/commits/38e4540c710c9f049678004c1df99e03f2de382f)